### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by gjuliao**
## Requirements

Insert Data through ruby console.

- Insert Data through ruby console.  ✅ 
- Create the following methods:

**Users**
- A method that returns the 3 most recent posts for a given user.
image

**Posts**

- A method that updates the posts counter for a user.
image

- A method which returns the 5 most recent comments for a given post.
image

**Comments**

- A method that updates the comments counter for a post.
image

**Likes**

- A method that updates the likes counter for a post.
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,12 @@
...
+  has_many :posts
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_count
+    self.posts_counter = posts.count
+    save
+  end
```
Comment (ishimwezachee) [inline app/models/user.rb]: - [ ]  The user model should have many comments and likes as well. As you know, it is the user who will create the comments and the likes, so you should have that relationship created as well.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+
+  def update_comments_count
+    self.comments_counter = comments.count
+    save
+  end
+
+  def update_likes_count
+    self.likes_counter = likes.count
+    save
+  end
```
Comment (ishimwezachee) [inline app/models/post.rb]: - [ ]  You should have added methods to their corresponding models. try to move update_comment_count to comment, update_like_count to like model as well.

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @gjuliao  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image


## Highlights

- Linters are passing ✅
- PR is well described ✅

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 2 above)
Comment (VICTOR-CHUKWU) [inline app/models/post.rb]: - [ ] [OPTIONAL AND IMPORTANT] since you have these two methods in their respective model, you can remove it from here

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @gjuliao , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**